foundry/glm/glm.py

```diff
@@ -314,10 +409,19 @@ def _get_ydict(self, y: ModelMatrix, sample_weight: Optional[np.ndarray]) -> Dic
                 raise NotImplementedError(
                     f"Unclear how to handle {k}, please report this error to the package maintainer"
                 )
+
+            # special handling for classification:
+            if k == 'value' and self.label_encoder_ is not None:
+                ydict['value'] = self.label_encoder_.transform(ydict['value'])
+                ydict['value'] = to_tensor(ydict['value'], dtype='int', device=_to_kwargs['device'])
```

Comment:
```suggestion
                ydict['value'] = to_tensor(ydict['value'], dtype=torch.long, device=_to_kwargs['device'])
```

I was getting an error. Unsure what this does, this modification seemed to let the code run.